src/frontend.c

```diff
@@ -126,6 +169,77 @@ int main(int argc, char *argv[]) {
   finish(0);
 }
 
+/* Handle keypresses before they are sent to modes.
+ *
+ * Returns, 0 if no characters were read, 1 if characters were read, and
+ * negative on error.
+ */
+int global_handler(WINDOW *win) {
+  int c = wgetch(win);
+  switch (c) {
+    case 18:  // ^R, read from savefile
```

Comment:
Try the following for cleaner switch statement 

```
#ifndef CTRL
#define CTRL(c) ((c) & 037)
#endif

switch(getch()) {
case CTRL('r'):
    /* key ctrl-r (i.e. ^R) pressed */
```

Reply:
Added to `util.h`